### ISSUE TITLE
Output of predict_many for SKL2RiverRegressor

### ISSUE THREAD

**Issue by yolking** [online-ml/river#1251]
Hello. I am mostly interested in `SGDRegressor` form `sklearn`, that's why I use `convert_sklearn_to_river(SGDRegressor())` in pipeline. Basically in docs it's said that `predict_one` would learn all unsupervised transformers and it works fine in case of `predict_one` method. But I want to use `predict_many` instead for batches.

So in case of `LinearRegression` I try
```py
from river import linear_model
from river import metrics
from river import stream
from river import compat
import numpy as np
from sklearn import datasets
from sklearn.linear_model import SGDRegressor

np.random.seed(1000)
X, y = datasets.make_regression(n_samples=500, n_features=4)
X = pd.DataFrame(X)
y = pd.DataFrame(y)
model = preprocessing.StandardScaler() | linear_model.LinearRegression(intercept_lr=.1)
model.predict_many(X)
```
I get
```
0      0.0
1      0.0
2      0.0
3      0.0
4      0.0
      ... 
495    0.0
496    0.0
497    0.0
498    0.0
499    0.0
Length: 500, dtype: float64
```

Now I try the same with SGDRegressor:
```py
np.random.seed(1000)
X, y = datasets.make_regression(n_samples=500, n_features=4)
X = pd.DataFrame(X)
y = pd.DataFrame(y)
model = preprocessing.StandardScaler() | compat.convert_sklearn_to_river(SGDRegressor())
model.predict_many(X)
```
I get
`NotFittedError: This SGDRegressor instance is not fitted yet. Call 'fit' with appropriate arguments before using this estimator.
`
This makes combined pipeline unusable and I have to use two separate pipelines instead for transformers and regressor.
But it seems it works fine if I use `predict_one` method:
```py
for x in X.to_dict('records'):
    print(model.predict_one(x))
```
Basically same check should be added as in [predict_one](https://github.com/online-ml/river/blob/201440b5129d6b66455ed0911038d5892ffd6343/river/compat/sklearn_to_river.py#L96)

**Comment by MaxHalford** [comment]
Good catch! This is now fixed on the `main` branch.
